File: Python3/9. Palindrome Number.py
class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        x_tostring = str(x)
        reverse_x_tostring = x_tostring[::-1]
        if x_tostring == reverse_x_tostring :
            return True
        else :
            return False

# 执行用时：380 ms
# 已经战胜 89.06 % 的 python3 提交记录

File: 9. Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        if (x < 10) return true;
        int rever = 0;
        int raw = x;
        while (x != 0) {
            rever *= 10;
            rever += (x % 10);
            x /= 10;
        }
        return rever == raw;
    }
};

// 执行用时：220 ms
// 已经战胜 37.36 % 的 cpp 提交记录